tests/keras/test_pyfunc_model_works_with_all_input_types.py

```diff
@@ -0,0 +1,158 @@
+# pep8: disable=E501
+
+
+from distutils.version import LooseVersion
+import os
+import pytest
+
+import keras
+from keras.models import Sequential, Model
+from keras.layers import Dense, Input, Concatenate
+from keras.optimizers import SGD
+import sklearn.datasets as datasets
+import pandas as pd
+import numpy as np
+
+import mlflow
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(tmpdir.strpath, "model")
+
+
+@pytest.fixture(scope="module")
+def data():
+    iris = datasets.load_iris()
+    data = pd.DataFrame(
+        data=np.c_[iris["data"], iris["target"]], columns=iris["feature_names"] + ["target"]
+    )
+    y = data["target"]
+    x = data.drop("target", axis=1)
+    return x, y
+
+
+@pytest.fixture(scope="module")
+def single_tensor_input_model(data):
+    x, y = data
+    model = Sequential()
+    model.add(Dense(3, input_dim=4))
+    model.add(Dense(1))
+    # Use a small learning rate to prevent exploding gradients which may produce
+    # infinite prediction values
+    lr = 0.001
+    kwargs = (
+        # `lr` was renamed to `learning_rate` in keras 2.3.0:
+        # https://github.com/keras-team/keras/releases/tag/2.3.0
+        {"lr": lr}
+        if LooseVersion(keras.__version__) < LooseVersion("2.3.0")
+        else {"learning_rate": lr}
+    )
+    model.compile(loss="mean_squared_error", optimizer=SGD(**kwargs))
+    model.fit(x.values, y.values)
+    return model
+
+
+@pytest.fixture(scope="module")
+def multi_tensor_input_model_list(data):
+    x, y = data
```

Comment:
nit:It probably does not matter too much for the test, but could we change the input to be 2 tensors of shape(-1,2) instead of (-1, 4) and pass the first 2 columns of the dataframe as input_a and the second 2 as tensor_b?

That way the model should look the same as in other case and it may be easier to check / reason about it if something goes wrong.